account_statement_import_coda/wizard/account_statement_import_coda.py

```diff
@@ -0,0 +1,218 @@
+            info = self.get_st_line_vals(line, globalisation_dict, information_dict)
+            info["sequence"] = sequence
+            info["unique_import_id"] = (
+                statement.coda_seq_number
+                + "-"
+                + statement.old_balance_date
+                + "-"
+                + statement.new_balance_date
+                + "-"
+                + info["unique_import_id"]
+            )
+            transactions.append(info)
+        return vals
+
+    def get_st_line_note(self, line, information_dict):
+        """This method returns a formatted note from line information"""
+        note = []
+        if line.counterparty_name:
+            note.append(_("Counter Party") + ": " + line.counterparty_name)
+        if line.counterparty_number:
+            note.append(_("Counter Party Account") + ": " + line.counterparty_number)
+        if line.counterparty_address:
+            note.append(_("Counter Party Address") + ": " + line.counterparty_address)
+        infos = information_dict.get(line.transaction_ref, [])
+        if line.communication or infos:
+            communications = []
+            if line.communication:
+                communications.append(line.communication)
+            for info in infos:
+                communications.append(info.communication)
+            note.append(_("Communication") + ": " + " ".join(communications))
+        return note and "\n".join(note) or None
+
+    def get_st_line_name(self, line, globalisation_dict):
+        """
+        This method must return a valid name for the statement line
+        The name is the statement communication if exists or
+        the communication of the related globalisation line if exists or
+        '/'
+        """
+        name = line.communication
+        if not name and line.ref_move in globalisation_dict:
+            name = globalisation_dict[line.ref_move].communication
+        return name or "/"
+
+    def get_st_line_vals(self, line, globalisation_dict, information_dict):
+        """
+        This method must return a dict of vals that can be passed to create
+        method of statement line in order to record it. It is the
+        responsibility of every parser to give this dict of vals,
+        so each one can implement his own way of recording the lines.
+            :param:  line: a dict of vals that represent a line of
+            result_row_list
+            :return: dict of values to give to the create method of
+            statement line,
+                     it MUST contain at least:
+                {
+                    'name':value,
+                    'date':value,
+                    'amount':value,
+                    'payment_ref':value,
+                }
+        """
+        amount = line.transaction_amount
+        if line.transaction_amount_sign == AmountSign.DEBIT:
+            amount = -amount
+        return {
+            "name": self.get_st_line_name(line, globalisation_dict),
```

Comment:
Now that account.bank.statement.line inherit from account.move, you're not supposed to force the "name" in this dict! Otherwise, the move will not have the "name" set by this code instead of having the "name" set by the journal like all the other moves. You should just set "payment_ref".

Reply:
Hi @alexis-via. I agree with you. A PR is opened since several months with this fix, but wasn't approved yet. See https://github.com/OCA/l10n-belgium/pull/139

Btw, a second PR with another fix is here, could you also have a look please? https://github.com/OCA/l10n-belgium/pull/140